smart-send-logistics/includes/frontend/class-ss-shipping-endpoint.php

```diff
@@ -0,0 +1,130 @@
+     *
+     * @return void
+     */
+    public function register_endpoints()
+    {
+        // Register the API endpoint
+        register_rest_route('smart-send-logistics/v1', '/get-closest-pickup-points', array(
+            'methods' => 'POST',
+            'callback' => array($this, 'get_pickup_points_callback'),
+            'permission_callback' => "__return_true",
+        ));
+        // Register the API endpoint for shipping carrier
+        register_rest_route('smart-send-logistics/v1', '/get-shipping-carrier', array(
+            'methods' => 'POST',
+            'callback' => array($this, 'get_shipping_carrier'),
+            'permission_callback' => "__return_true",
+        ));
+    }
+
+    /**
+     * Callback function for the REST API endpoint to fetch pickup points.
+     *
+     * @param WP_REST_Request $request The request object containing parameters.
+     * @return WP_REST_Response Response object with pickup points or an error message.
+     */
+    public function get_pickup_points_callback(WP_REST_Request $request)
+    {
+        $country = wc_clean($request->get_param('country'));
+        $postal_code = wc_clean($request->get_param('postCode'));
+        $city = (!empty($request->get_param('city')) ? wc_clean($request->get_param('city')) : null);
+        $street = wc_clean($request->get_param('street'));
+        $carrier = wc_clean($request->get_param('carrier'));
+        $is_rest_api = true;
+
+        $ss_agents = $this->frontend->handle_agents_session($carrier, $country, $postal_code, $city, $street, $is_rest_api);
+        if (!empty($ss_agents)) {
+            $ss_agent_options = array();
+            foreach ($ss_agents as $key => $agent) {
+                $formatted_address = $this->frontend->get_formatted_address_for_endpoint($agent);
+            
+                $ss_agent_options[$agent->agent_no] = $formatted_address;
+            }
+
+
+            return new WP_REST_Response($ss_agents, 200);
+        } else {
+            return new WP_REST_Response(array('message' => __('No pick-up points found', 'smart-send-logistics')), 404);
+        }
+    }
+
+    /**
+     * Get shipping method meta data
+     */
+    public function get_shipping_carrier(WP_REST_Request $request)
+    {
+        // Get the shipping method value from the request
+        $shipping_method = wc_clean($request->get_param('ss_method'));
+        if (empty($shipping_method)) {
+            return new WP_REST_Response(array('message' => __('Invalid Shipping Method ID', 'smart-send-logistics')), 400);
+        }
+
+        $carrier_keys = ["name", "id"];
+
+        $shipping_method_parts = explode(":", $shipping_method);
+
+        if (count($shipping_method_parts) > 1) {
+            // Get the second part of the shipping method which is the ID
+            $shipping_method_parts = array_combine($carrier_keys, $shipping_method_parts);
+        }
+
+        $shipping_method_id = $shipping_method_parts["id"];
```

Comment:
Please add examples of how `$shipping_method` can look in the situation where count === 1 and where count > 1

Reply:
When  count === 1`, it handles cases where the selected shipping method value contains either the ID or the name, but not both, separated by a colon. When `count > 1,` it applies to cases where the shipping method value includes both the ID and name, separated by a colon.